### PR TITLE
Dbrefresh with node

### PR DESCRIPTION
**Pull request by clemarescx**
## What does this do?
Runs the database initialisation / restore script with NodeJS instead of *nix shell to make the script cross-platform (thus now including Windows).

The new script looks in the `src` directory for all JSON files with names following the pattern *5e-SRD-**\<datatype\>**.json*, where **\<datatype\>** is the collection name in Camel-Kebab case (e.g.: Ability-Scores) and the mongodb collection name is generated by setting **\<datatype\>** to lowercase. 

## How was it tested?
This was tested on Windows 10 with NodeJS v12.18.3 and MongoDB v4.2
MONGODB_URI was always set to the example URI: `mongodb://localhost/5e-database`

## Is there a Github issue this is resolving?
Not that I could see at first glance over issue titles 🙂

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6467 @@
+{
```
Comment (bagelbits) [inline package-lock.json]: Please use `yarn` instead of `npm`. There is already a yarn lockfile.

File: scripts/db-refresh.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+} catch (e) {
+    console.error('could not execute mongoimport - make sure the directory containing mongoimport is visible in your $PATH environment variable')
+    process.exit(1);
+}
+
+const json_db_dir = 'src';
+const json_db_collection_prefix = '5e-SRD-';
+const json_data_pattern = `\\b${json_db_collection_prefix}(.+)\.json\\b`;
+
+const regex = new RegExp(json_data_pattern);
```
Comment (bagelbits) [inline scripts/db-refresh.js:32]: This will end up picking up the Test file. Though in all honesty, I'm not even sure why that file exists.

Comment (bagelbits) [reply scripts/db-refresh.js:32]: I might just remove the file, honestly.

Comment (bagelbits) [reply scripts/db-refresh.js:32]: I made a separate PR to fix this: https://github.com/bagelbits/5e-database/pull/227

**Review by bagelbits — CHANGES_REQUESTED** [review]
Really just the switch from `npm` to `yarn` needs to change. Otherwise, this looks fine!

File: package-lock.json (hunk 1 above)
Comment (clemarescx) [reply package-lock.json]: Sorry I'm not sure what you meant by that, did you mean I should delete the *package-lock.json* file?

File: scripts/db-refresh.js (hunk 2 above)
Comment (clemarescx) [reply scripts/db-refresh.js:32]: I edited the PR description text to remove the mention of the *test* JSON file + collection.

File: package-lock.json (hunk 1 above)
Comment (bagelbits) [reply package-lock.json]: Yes please! Sorry, I thought you also ended up installing new packages as well. But it looks like you weren't.

Comment (bagelbits) [reply package-lock.json]: Or apparently, I can?

**Review by bagelbits — APPROVED** [review]
LGTM!

**Pull request merged**